sound/soc/sof/topology.c

```diff
@@ -2142,9 +2188,24 @@ static int sof_link_hda_load(struct snd_soc_component *scomp, int index,
 	if (link->dpcm_capture) {
 		ret = sof_link_hda_process(sdev, link, config, rx_slot,
 					   SNDRV_PCM_STREAM_CAPTURE);
-		if (ret < 0)
+		if (ret < 0) {
 			dev_err(sdev->dev, "error: failed to set DAI config for capture dai HDA%d\n",
 				config->dai_index);
+
+			return ret;
+		}
+	}
+
+	/* set config for all DAI's with name matching the link name */
+	ret = sof_set_dai_config(sdev, size, link, config);
+	if (ret < 0) {
+		if (link->dpcm_playback)
+			dev_err(sdev->dev, "error: failed to save DAI config for playback HDA%d\n",
+				config->dai_index);
+
+		if (link->dpcm_capture)
+			dev_err(sdev->dev, "error: failed to save DAI config for capture HDA%d\n",
+				config->dai_index);
```

Comment:
Hum, wondering if I am missing something. Why do you call out playback and capture for HDAudio but not for SSP? Is this intended? Is it needed, if yes should we have this for SSP as well?